docs/next.config.js

```diff
@@ -17,21 +17,31 @@ module.exports = withCSS({
   // Rather than use `@zeit/next-mdx`, we replicate it
   pageExtensions: ['js', 'jsx', 'md', 'mdx'],
   webpack: (config, options) => {
-    // Create a copy of the babel loader, to separate MDX and Next/Preval caches
-    const babelMdxLoader = {
-      ...options.defaultLoaders.babel,
-      options: {
-        ...options.defaultLoaders.babel.options,
-        cacheDirectory: 'node_modules/.cache/babel-mdx-loader',
+    // Add preval support for `constants/*` only and move it to the `.next/preval` cache.
+    // It's to prevent over-usage and separate the cache to allow manually invalidation.
+    // See: https://github.com/kentcdodds/babel-plugin-preval/issues/19
+    config.module.rules.push({
+      test: /.jsx?$/,
+      include: [resolve(__dirname, 'constants')],
+      use: {
+        ...options.defaultLoaders.babel,
+        options: {
+          ...options.defaultLoaders.babel.options,
+          cacheDirectory: '.next/preval',
+          plugins: ['preval'],
```

Comment:
needed?
```suggestion
          plugins: [...options.defaultLoaders.babel.options?.plugins ?? [], 'preval'],
```

Reply:
Good point, I don't think we use any special plugins here. It's not rendered by Next or has functionality other than defining variables. But I agree, it's good dev hygiene 😄